Broken project when `project_name` != `project_slug`

## Expected Behavior

If a new project is created with `project_name` distinct from `project_slug` it should be possible to run the project tests.

## Actual Behavior

If a new project is created with `project_name` distinct from `project_slug` the tests are broken.

## Steps to Reproduce the Problem

  1. Create a new project:
```
$ cookiecutter gh:fedejaure/cookiecutter-modern-pypackage --checkout v2.0.0
full_name [Federico Jaureguialzo]: 
email [fedejaure@example.com]: 
github_username [fedejaure]: 
project_name [modern-python-boilerplate]: 
project_slug [modern_python_boilerplate]: modern_python
project_title [Modern Python Boilerplate]: 
project_short_description [Python Boilerplate contains all the boilerplate you need to create a modern Python package.]: 
version [0.1.0]: 
Select open_source_license:
1 - MIT
2 - BSD
3 - ISC
4 - Apache Software License 2.0
5 - GNU General Public License v3
6 - Not open source
Choose from 1, 2, 3, 4, 5, 6 [1]: 
Select command_line_interface:
1 - Click
2 - No command-line interface
Choose from 1, 2 [1]: 
add_code_of_conduct [y]: 
add_contributing_file [y]: 
add_security_file [y]: 
contact_method:
```
  2. Go to the new project dir and install the deps:
```
$ cd modern-python-boilerplate
modern-python-boilerplate $ poetry install
```
  3. Run the tests:
```
modern-python-boilerplate $ poetry run inv tests 
============================================================================================================ test session starts =============================================================================================================
platform darwin -- Python 3.9.1, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /modern-python-boilerplate
plugins: cov-2.12.1, xdoctest-0.15.5
collected 0 items / 2 errors                                                                                                                                                                                                                 
Coverage.py warning: Module modern_python was never imported. (module-not-imported)
Coverage.py warning: No data was collected. (no-data-collected)
WARNING: Failed to generate report: No data to report.

/Caches/pypoetry/virtualenvs/modern-python-boilerplate-E8bi7dWk-py3.9/lib/python3.9/site-packages/pytest_cov/plugin.py:271: PytestWarning: Failed to generate report: No data to report.

  self.cov_controller.finish()

=================================================================================================================== ERRORS ===================================================================================================================
_____________________________________________________________________________________________________ ERROR collecting tests/test_cli.py _____________________________________________________________________________________________________
ImportError while importing test module '/modern-python-boilerplate/tests/test_cli.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/Cellar/python@3.9/3.9.1_3/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_cli.py:7: in <module>
    import modern_python
E   ModuleNotFoundError: No module named 'modern_python'
________________________________________________________________________________________________ ERROR collecting tests/test_modern_python.py ________________________________________________________________________________________________
ImportError while importing test module '/modern-python-boilerplate/tests/test_modern_python.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/Cellar/python@3.9/3.9.1_3/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_modern_python.py:6: in <module>
    import modern_python
E   ModuleNotFoundError: No module named 'modern_python'


========================================================================================================== short test summary info ===========================================================================================================
ERROR tests/test_cli.py
ERROR tests/test_modern_python.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================================================================================= 2 errors in 0.19s ==============================================================================================================
```

## Specifications

  - Version: v2.0.0

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -28,7 +28,7 @@ jobs:
 
   tests:
     needs: linting
-    name: Python ${{ matrix.python-version }}
+    name: ${{ matrix.os }} / ${{ matrix.python-version }}
     runs-on: ${{ matrix.os }}-latest
     strategy:
       matrix:

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -29,9 +29,10 @@ jobs:
   tests:
     needs: linting
     name: Python ${{ matrix.python-version }}
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}-latest
     strategy:
       matrix:
+        os: [Ubuntu, MacOS]
         python-version: [3.7, 3.8, 3.9]
       fail-fast: true
     steps:

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 
 ## [Unreleased]
+### Fixed
+- package discovery when `project_name` != `project_slug`
 
 ## [2.0.0] - 2021-06-12
 ### Added

--- a/cookiecutter.json
+++ b/cookiecutter.json
@@ -3,7 +3,7 @@
   "email": "fedejaure@example.com",
   "github_username": "fedejaure",
   "project_name": "modern-python-boilerplate",
-  "project_slug": "{{ cookiecutter.project_name.lower().replace('-', '_').replace('-', '_') }}",
+  "project_slug": "{{ cookiecutter.project_name.lower().replace('-', '_') }}",
   "project_title": "{{ cookiecutter.project_name.replace('-', ' ').title() }}",
   "project_short_description": "Python Boilerplate contains all the boilerplate you need to create a modern Python package.",
   "version": "0.1.0",

--- a/tests/test_bake_project.py
+++ b/tests/test_bake_project.py
@@ -17,6 +17,10 @@ COOKIE_CONTEXT_NOT_OPEN_SOURCE = {"open_source_license": "Not open source"}
 COOKIE_CONTEXT_CLI = {"command_line_interface": "Click"}
 COOKIE_CONTEXT_BSD = {"open_source_license": "BSD"}
 COOKIE_CONTEXT_NO_CLI = {"command_line_interface": "No command-line interface"}
+COOKIE_CONTEXT_DIFF_NAME_AND_SLUG = {
+    "project_name": "some-project-name",
+    "project_slug": "some_project",
+}
 
 
 @contextmanager
@@ -123,6 +127,7 @@ def _test_bake_and_run_invoke_tasks(
         COOKIE_CONTEXT_NOT_OPEN_SOURCE,
         {**COOKIE_CONTEXT_NOT_OPEN_SOURCE, **COOKIE_CONTEXT_NO_CLI},
         {**COOKIE_CONTEXT_CLI, **COOKIE_CONTEXT_BSD},
+        COOKIE_CONTEXT_DIFF_NAME_AND_SLUG,
     ],
 )
 def test_bake_and_run_invoke(cookies: Cookies, extra_context: Dict[str, str]) -> None:
@@ -134,20 +139,6 @@ def test_bake_and_run_invoke(cookies: Cookies, extra_context: Dict[str, str]) ->
         "tests",
         "coverage",
         "version -d minor",
+        "docs",
     ]
     _test_bake_and_run_invoke_tasks(cookies, extra_context, invoke_tasks)
-
-
-@skip_on_windows
-@pytest.mark.parametrize(
-    "extra_context",
-    [
-        {},
-        COOKIE_CONTEXT_NOT_OPEN_SOURCE,
-        {**COOKIE_CONTEXT_NOT_OPEN_SOURCE, **COOKIE_CONTEXT_NO_CLI},
-        {**COOKIE_CONTEXT_CLI, **COOKIE_CONTEXT_BSD},
-    ],
-)
-def test_bake_and_build_docs(cookies: Cookies, extra_context: Dict[str, str]) -> None:
-    """Test bake the project and check invoke docs task."""
-    _test_bake_and_run_invoke_tasks(cookies, extra_context, ["docs"])

--- a/{{cookiecutter.project_name}}/pyproject.toml
+++ b/{{cookiecutter.project_name}}/pyproject.toml
@@ -30,7 +30,11 @@ classifiers=[
     "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
 ]
-
+{% if cookiecutter.project_name.lower().replace('-', '_') != cookiecutter.project_slug %}
+packages = [
+    { include = "{{ cookiecutter.project_slug }}", from = "src" },
+]
+{% endif %}
 
 [tool.poetry.urls]
 "Bug Tracker" = "https://github.com/{{ cookiecutter.github_username }}/{{ cookiecutter.project_name }}/issues"

